tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/field_access_negative.bal

```diff
@@ -376,3 +376,25 @@ function testInvalidBoundMethodAccessWithRemoteMethod(ServiceClass a,
 
     _ = e.ser.fn;
 }
+
+function testInvalidXMLMapFieldAccess1() returns error? {
+    map<xml> m = {a: xml `foo`};
+    xml x = check m.a; // error
+}
+
+function testInvalidXMLMapFieldAccess2() returns error? {
+    map<xml> m = {a: xml `foo`};
+    xml x = check m.b; // error
+}
+
+function testInvalidXMLMapFieldAccess3() returns error? {
+    map<xml> m = {};
+    m["a"] = xml `foo`;
+    xml x = check m.a; // error
+}
+
+function testInvalidXMLMapFieldAccess4() returns error? {
+    map<xml|json> m = {};
+    m["a"] = xml `foo`;
+    xml|json x = check m.a; // error
+}
```

Comment:
Do we have tests for optional field access like
```

function testInvalidXMLMapFieldAccess1() returns error? {
    map<xml> m = {a: xml `foo`};
    _ = m?.a;
}
```

if not shall we few tests?

Reply:
added.